test/integration/ads_integration_test.cc

```diff
@@ -1749,149 +1749,172 @@ TEST_P(AdsIntegrationTest, ContextParameterUpdate) {
+  const auto listeners = {
+      buildListener("xdstp://test/envoy.config.listener.v3.Listener/foo-listener/"
+                    "bar?xds.node.cluster=cluster_name&xds.node.id=node_name",
+                    route_name_0),
+      // Ignore resource in impostor namespace.
+      buildListener("xdstp://test/envoy.config.listener.v3.Listener/impostor/"
+                    "bar?xds.node.cluster=cluster_name&xds.node.id=node_name",
+                    route_name_0),
+      // Ignore non-matching context params.
+      buildListener("xdstp://test/envoy.config.listener.v3.Listener/foo-listener/"
+                    "baz?xds.node.cluster=cluster_name&xds.node.id=other_name",
+                    route_name_0),
+  };
+  sendDiscoveryResponse<envoy::config::listener::v3::Listener>(Config::TypeUrl::get().Listener,
+                                                               listeners, listeners, {}, "1");
+  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().ClusterLoadAssignment, "1", {}, {}, {},
+                                      /*expect_node=*/isSotw()));
+
+  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().RouteConfiguration, "", {route_name_0},
+                                      {route_name_0}, {}, /*expect_node=*/isSotw()));
+  const auto route_config = buildRouteConfig(route_name_0, cluster_name);
   sendDiscoveryResponse<envoy::config::route::v3::RouteConfiguration>(
-      Config::TypeUrl::get().RouteConfiguration, {}, {buildRouteConfig(route_name_0, cluster_name)},
-      {}, "1");
+      Config::TypeUrl::get().RouteConfiguration, {route_config}, {route_config}, {}, "1");
 
-  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().Listener, "1", {}, {}, {}));
-  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().RouteConfiguration, "1", {}, {}, {}));
+  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().Listener, "1", {}, {}, {},
+                                      /*expect_node=*/isSotw()));
+  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().RouteConfiguration, "1", {}, {}, {},
+                                      /*expect_node=*/isSotw()));
 
   test_server_->waitForCounterEq("listener_manager.listener_create_success", 1);
   makeSingleRequest();
 
   // Add a second listener in the foo namespace.
+  const auto second_listener =
+      buildListener("xdstp://test/envoy.config.listener.v3.Listener/foo-listener/"
+                    "baz?xds.node.cluster=cluster_name&xds.node.id=node_name",
+                    route_name_1);
   sendDiscoveryResponse<envoy::config::listener::v3::Listener>(
-      Config::TypeUrl::get().Listener, {},
-      {buildListener("xdstp://test/envoy.config.listener.v3.Listener/foo-listener/"
-                     "baz?xds.node.cluster=cluster_name&xds.node.id=node_name",
-                     route_name_1)},
-      {}, "2");
+      Config::TypeUrl::get().Listener, {second_listener}, {second_listener}, {}, "2");
```

Comment:
To keep consistency between SotW and Delta, I believe the 2nd argument needs to pass all the relevant listeners.